wire-ios-data-model/Source/Model/UserClient/E2eIdentityCertificate.swift

```diff
@@ -0,0 +1,69 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import X509
+import SwiftASN1
+import WireCoreCrypto
+
+public enum E2EIdentityCertificateStatus: CaseIterable {
+    case notActivated, revoked, expired, valid
+}
+
+public struct E2eIdentityCertificate: Equatable {
+    public var details: String
+    public var mlsThumbprint: String
+    public var notValidBefore: Date
+    public var expiryDate: Date
+    public var status: E2EIdentityCertificateStatus
+    public var serialNumber: String
+
+    public init(
+        certificateDetails: String,
+        mlsThumbprint: String,
+        notValidBefore: Date,
+        expiryDate: Date,
+        certificateStatus: E2EIdentityCertificateStatus,
+        serialNumber: String
+    ) {
+        self.details = certificateDetails
+        self.mlsThumbprint = mlsThumbprint
+        self.notValidBefore = notValidBefore
+        self.expiryDate = expiryDate
+        self.status = certificateStatus
+        self.serialNumber = serialNumber
+    }
+
+    public init(
+           certificate: Certificate,
+           certificateDetails: String,
+           certificateStatus: E2EIdentityCertificateStatus,
+           mlsThumbprint: String
+       ) {
+           self.details = certificateDetails
+           self.notValidBefore = certificate.notValidBefore
+           self.expiryDate = certificate.notValidAfter
+           self.serialNumber = certificate.serialNumber.description.replacingOccurrences(of: ":", with: "")
```

Comment:
The `description` property should not be accessed directly. See Apple [docs](https://developer.apple.com/documentation/swift/customstringconvertible):
```
...
The String(describing:) initializer is the preferred way to convert an instance of any type to a string.
...
Accessing a type’s description property directly ... is discouraged.
```